packages/wonder-blocks-data/util/types.js

```diff
@@ -1,4 +1,16 @@
 // @flow
+export type Result<TData> =
+    | {|
+          loading: true,
+          data?: void,
+          error?: void,
+      |}
+    | {|
+          loading: false,
+          data?: TData,
+          error?: string,
```

Comment:
There won't be a case where both error and data are defined, correct?

Reply:
That's correct. I tried typing it to make that clear, but hit confusing flow errors.